src/data_info/services/sharing.clj

```diff
@@ -160,10 +161,15 @@
     (when-not (nil? matching-key)
       (string/replace-first path matching-key (mappings matching-key)))))
 
+(defn- encode-mapped-anon-path
+  "Take a path that's been appropriately mapped, and URL encode it for use. This function will use %20 for spaces."
+  [mapped-path]
+  (string/join "/" (map #(string/replace (URLEncoder/encode %) #"\+" "%20") (string/split mapped-path #"/"))))
```

Comment:
Looks good to me. I might have used `as->` threading here, but this is also easy enough to follow, I think.

Reply:
I like that idea, I switched it over to using that.